source/extensions/access_loggers/fluentd/fluentd_access_log_impl.h

```diff
@@ -0,0 +1,162 @@
+#pragma once
+
+#include <chrono>
+
+#include "envoy/extensions/access_loggers/fluentd/v3/fluentd.pb.h"
+#include "envoy/extensions/access_loggers/fluentd/v3/fluentd.pb.validate.h"
+
+#include "source/common/formatter/substitution_formatter.h"
+#include "source/extensions/access_loggers/common/access_log_base.h"
+#include "source/extensions/access_loggers/fluentd/substitution_formatter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace AccessLoggers {
+namespace Fluentd {
+
+using FluentdAccessLogConfig =
+    envoy::extensions::access_loggers::fluentd::v3::FluentdAccessLogConfig;
+using FluentdAccessLogConfigSharedPtr = std::shared_ptr<FluentdAccessLogConfig>;
+
+// Entry represents a single Fluentd message, msgpack format based, as specified in:
+// https://github.com/fluent/fluentd/wiki/Forward-Protocol-Specification-v1#entry
+class Entry {
+public:
+  Entry(const Entry&) = delete;
+  Entry& operator=(const Entry&) = delete;
+  Entry(uint64_t time, std::vector<uint8_t>&& record) : time_(time), record_(record) {}
+
+  uint64_t time_;
```

Comment:
nit: should probably mark entry fields with const

Reply:
done, thanks